subsys/mgmt/mcumgr/transport/smp_bt.c

```diff
@@ -493,7 +514,35 @@ int smp_bt_unregister(void)
 static void connected(struct bt_conn *conn, uint8_t err)
 {
 	if (err == 0) {
-		conn_param_data_alloc(conn);
+		struct conn_param_data *cpd;
+		bool acquired = false;
+		uint8_t i;
+
+		cpd = conn_param_data_alloc(conn);
+		k_sem_reset(&cpd->smp_notify_sem);
+
+		/* Generate an ID for this connection */
```

Comment:
What are the chances that `connected` gets preempted and other `connected` instance gets called?
We can use CPU ticks here or something like that, there is very low chance that a connection will be held long enough to wrap around and get the same ticks for two different connections, and even in code executed in parallel, unless two CPUs are used, ticks should differ.
We can probably use only the lower 32bits of  `k_uptime_ticks(void)` safely.

Reply:
> I have some trouble following why this ID is needed, could you elaborate?
The ID is needed because a connection can be established, a mcumgr command can be sent, then that device can disconnect and a new device can connect - if this is a long running command it can take seconds to run, at which point without the ID check it will send the response to the new device which is invalid operation. In terms of a race condition, this is executed in the system workqueue, which runs to completion, and would need to exhaust 255 entries to loop around (which, assuming someone was connecting and disconnecting continuously, assuming it would take 30ms to do a cycle, that would take 7.65 seconds)

> What are the chances that connected gets preempted and other connected instance gets called
0 as far as I'm aware as it's called in the system workqueue which is blocked until it has completed (co-operative thread)